strawberry/annotation.py

```diff
@@ -131,6 +146,8 @@ def set_namespace_from_field(self, field: StrawberryField) -> None:
         module = sys.modules[field.origin.__module__]
         self.namespace = module.__dict__
 
+        self.__eval_cache__ = None  # Invalidate cache to allow re-evaluation
```

Comment:
why is this needed?

Reply:
To keep it consistent, probably there isn't a use-case for it. But if the
field somehow changes and it lives in a different module, then the previously
evaluated type could be incorrect